### PR TITLE
Add PROTOBUF_NOSR

### PR DESCRIPTION
**Pull request by Gerrrr**
### Description 

This patch adds a new format `PROTOBUF_NOSR` and fixes a bug in qtt tests where multi-format tests always used `JSON`.

The new `PROTOBUF_NOSR` format allows to serde Protobuf messages without using Schema Registry. Instead, it creates the internal Protobuf schema from the logical schema. The limitation of this format is that it does not support dependencies to other schemas.

**NB** this format works with Protobuf messages serialized into bytes. As a result, it won't work with `io.confluent.kafka.serializers.protobuf.KafkaProtobufSerializer` that has a custom serialization format with a magic byte, SR id, and message indexes before the payload.

This patch introduces `ProtobufNoSRSerdeFactory` that differs from `ProtobufSerdeFactory` in the following ways:
* `createSerde` does not accept `Supplier<SchemaRegistryClient>` as one of its arguments.
* `getConverter` uses `ProtobufNoSRConverter` instead of `ProtobufConverter`

`ProtobufNoSRConverter` differs from `ProtobufConverter` in the following ways:
* The constructor does not accept `SchemaRegistryClient` as an argument
* The serializer and deserializer do not extend `AbstractKafkaProtobufSerializer`. Instead, they do direct serde to/from a byte array.

While adding the new format to qtt tests, I discovered that multiple multi-format tests always use `JSON`. Some examples are:
* [url.json](https://github.com/confluentinc/ksql/pull/9078/files#diff-530c53cb57811b33ff9f73b7a92cd5cb2a98cb2f92213dbeeb3f213763223a41R11)
* [initcap.json](ksqldb-functional-tests/src/test/resources/query-validation-tests/initcap.json)

### Testing done 

Unit, functional, manual tests.

Specifically, I ran 2 manual tests:

#### ksqlDB reads Protobuf messages produced from Java

1. Create a Protobuf message:

```protobuf
syntax = "proto3";

package v2;
option java_package = "com.github.simplesteph.kafka.apps.v2";


message Person {
  string name = 1;
  int32 id = 2;
  string email = 3;

  enum PhoneType {
    MOBILE = 0;
    HOME = 1;
    WORK = 2;
  }

  message PhoneNumber {
    string number = 1;
    PhoneType type = 2;
  }

  repeated PhoneNumber phones = 4;
}
```

2. Generate a Java class from the message.
3.  Add a number of entries to a topic:

```java
for (int i = 0; i < 10; i++) {
            PhoneNumber number0 = PhoneNumber.newBuilder()
                .setNumber("123-456-78" + i)
                .setType(PhoneType.HOME)
                .build();
            PhoneNumber number1 = PhoneNumber.newBuilder()
                .setNumber("987-654-32" + i)
                .setType(PhoneType.MOBILE)
                .build();
            Person person = Person.newBuilder()
                .setId(i)
                .setEmail(i + "me@gmail.com")
                .setName("Name Name" + i)
                .addPhones(number0)
                .addPhones(number1)
                .build();

            ProducerRecord<Integer, Person> producerRecord = new ProducerRecord<>(
                topic, i, person
            );

            producer.send(producerRecord, (metadata, exception) -> {
                if (exception == null) {
                    System.out.println(metadata);
                } else {
                    exception.printStackTrace();
                }
            });
        }
```

Custom ProtobufSerializer:

```
public class PlainProtobufSerializer<T extends Message> implements Serializer<T> {

        public PlainProtobufSerializer() {
        }

        @Override
        public void configure(Map<String, ?> configs, boolean isKey) {

        }

        @Override
        public byte[] serialize(String topic, T data) {
            ByteArrayOutputStream out = new ByteArrayOutputStream();
            try {
                data.writeTo(out);
                out.close();
            } catch (IOException e) {
                e.printStackTrace();
            }
            return out.toByteArray();
        }

        @Override
        public void close() {

        }
    }
```

4. Create a ksql stream

```sql
ksql> CREATE STREAM persons (key INT KEY, name STRING, id INT, email STRING, phones ARRAY<STRUCT<number STRING, type INT>>) with (kafka_topic='persons', partitions
=1, value_format='PROTOBUF_NOSR');

 Message
----------------
 Stream created
----------------
```

5. Run a query against the stream:

```sql
ksql> select * from persons;
+------------------------------+------------------------------+------------------------------+------------------------------+------------------------------+
|KEY                           |NAME                          |ID                            |EMAIL                         |PHONES                        |
+------------------------------+------------------------------+------------------------------+------------------------------+------------------------------+
|0                             |Name Name0                    |0                             |0me@gmail.com                 |[{NUMBER=123-456-780, TYPE=1},|
|                              |                              |                              |                              | {NUMBER=987-654-320, TYPE=0}]|
|1                             |Name Name1                    |1                             |1me@gmail.com                 |[{NUMBER=123-456-781, TYPE=1},|
|                              |                              |                              |                              | {NUMBER=987-654-321, TYPE=0}]|
|2                             |Name Name2                    |2                             |2me@gmail.com                 |[{NUMBER=123-456-782, TYPE=1},|
|                              |                              |                              |                              | {NUMBER=987-654-322, TYPE=0}]|
|3                             |Name Name3                    |3                             |3me@gmail.com                 |[{NUMBER=123-456-783, TYPE=1},|
|                              |                              |                              |                              | {NUMBER=987-654-323, TYPE=0}]|
|4                             |Name Name4                    |4                             |4me@gmail.com                 |[{NUMBER=123-456-784, TYPE=1},|
|                              |                              |                              |                              | {NUMBER=987-654-324, TYPE=0}]|
|5                             |Name Name5                    |5                             |5me@gmail.com                 |[{NUMBER=123-456-785, TYPE=1},|
|                              |                              |                              |                              | {NUMBER=987-654-325, TYPE=0}]|
|6                             |Name Name6                    |6                             |6me@gmail.com                 |[{NUMBER=123-456-786, TYPE=1},|
|                              |                              |                              |                              | {NUMBER=987-654-326, TYPE=0}]|
|7                             |Name Name7                    |7                             |7me@gmail.com                 |[{NUMBER=123-456-787, TYPE=1},|
|                              |                              |                              |                              | {NUMBER=987-654-327, TYPE=0}]|
|8                             |Name Name8                    |8                             |8me@gmail.com                 |[{NUMBER=123-456-788, TYPE=1},|
|                              |                              |                              |                              | {NUMBER=987-654-328, TYPE=0}]|
|9                             |Name Name9                    |9                             |9me@gmail.com                 |[{NUMBER=123-456-789, TYPE=1},|
|                              |                              |                              |                              | {NUMBER=987-654-329, TYPE=0}]|
Query Completed
Query terminated
```

#### External consumer reads Protobuf messages produced by ksqlDB

1. Insert into topic from ksql:

```sql
ksql> insert into persons (key, name, id, email, phones) values (10, 'from-ksql', 10, 'from-ksql@mymail.com', ARRAY[STRUCT(number := '999-999-999', type := 2)]);
```

2. Read from java consumer:

```java
KafkaConsumer<Integer, Person> kafkaConsumer = new KafkaConsumer<>(properties);
        String topic = "persons";
        kafkaConsumer.subscribe(Collections.singleton(topic));

        while (true){
            ConsumerRecords<Integer, Person> records = kafkaConsumer.poll(1000);

            for (ConsumerRecord<Integer, Person> record : records){
                Person person = record.value();
                System.out.println(person);
            }

            kafkaConsumer.commitSync();
        }
```

Deserializer:

```java
public class PlainProtobufDeserializer implements Deserializer<Person> {

        public PlainProtobufDeserializer() {
        }

        @Override
        public void configure(Map<String, ?> configs, boolean isKey) {

        }

        @Override
        public Person deserialize(String topic, byte[] data) {
            if (data == null) {
                return null;
            }

            try {
                return PersonOuterClass.Person.parseFrom(data);
            } catch (InvalidProtocolBufferException e) {
                e.printStackTrace();
            }
            return null;
        }

        @Override
        public void close() {

        }
    }
```

Output (only last record for brevity):

```
name: "from-ksql"
id: 10
email: "from-ksql@mymail.com"
phones {
  number: "999-999-999"
  type: WORK
}
```

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Comment by Gerrrr** [comment]
Green CI - https://jenkins.confluent.io/job/Confluent%20Public%20Repo%20PR%20builder/job/ksql/job/PR-9078/4/.

**Review by cprasad1 — APPROVED** [review]
LGTM. Thanks @Gerrrr !

**Comment by cprasad1** [comment]
I think you would need to sign the agreement before it lets you merge. Unsure why we have to sign it again

**Comment by spena** [comment]
Wouldn't it be better to continue using the `PROTOBUF` format but just disable SR usage? Perhaps adding a new property in the WITH clause to do that. It can be reused for `AVRO` in the future.

**Review by agavra — APPROVED** [review]
LGTM - there's a good chunk of code duplication that would be nice to avoid, but also not the end of the world if we want to keep it straightforward.

Nice catch fixing the multi-format tests!

**Comment by Gerrrr** [comment]
Green CI https://jenkins.confluent.io/job/Confluent%20Public%20Repo%20PR%20builder/job/ksql/job/PR-9078/6/

File: ksqldb-serde/src/main/java/io/confluent/ksql/serde/protobuf/ProtobufNoSRConverter.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+import java.util.Map;
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.common.errors.InvalidConfigurationException;
+import org.apache.kafka.common.errors.SerializationException;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaAndValue;
+import org.apache.kafka.connect.errors.DataException;
+import org.apache.kafka.connect.storage.Converter;
+
+public class ProtobufNoSRConverter implements Converter {
```
[the remark this answers is not in the document]
Comment (Gerrrr) [reply ksqldb-serde/src/main/java/io/confluent/ksql/serde/protobuf/ProtobufNoSRConverter.java:39]: I've reduced the code duplication in [a435be0](https://github.com/confluentinc/ksql/pull/9078/commits/a435be0d8cb7f838c3f39e1be8e278e10ad8faa9).

**Pull request merged**